### ISSUE TITLE
marked particles in one output didn't correspond to the same particles

### ISSUE THREAD

**Issue by trquinn** [N-BodyShop/tipsy#3]
Date: Wed, 18 Nov 2009 14:01:08 -0800
From: Charlotte Christensen christensen@astro.washington.edu
To: Tom Quinn trq@astro.washington.edu
Subject: tipsy error

```
[ The following text is in the "ISO-8859-1" character set. ]
[ Your display is set for the "ANSI_X3.4-1968" character set.  ]
[ Some characters may be displayed incorrectly. ]
```

Hi Tom,
   This is the minor bug I found (or at least I think it is a bug).  I was
using writemark and readmark to track the same particles at different times.
I found that my marked particles in one output didn't correspond to the same
particles at a different timestep.  I think the code in readmark.c at lines
82 to 89 should be

else if (index < nsph + ndark && i < header.nsph + header.ndark)
     {
        mark_dark[i - nsph]++;
     }
else if(index < nbodies && i < header.nbodies)
     {
        mark_star[i - nsph - ndark]++;
     }

instead of

else if (index < nsph + ndark && i < header.nsph + header.ndark)
     {
        mark_dark[i - header.nsph]++;
     }
else if(index < nbodies && i < header.nbodies)
     {
        mark_star[i - header.nsph - header.ndark]++;
     }

Basically, nsph and hdark rather than header.nsph and header.ndark in the
index.

Charlotte
